chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/ProcessPeaksByPeakClassifierFilter.java

```diff
@@ -0,0 +1,169 @@
+import org.eclipse.chemclipse.processing.filter.CRUDListener;
+import org.eclipse.chemclipse.processing.filter.Filter;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.osgi.service.component.annotations.Component;
+
+@Component(service = { IPeakFilter.class, Filter.class, Processor.class })
+public class ProcessPeaksByPeakClassifierFilter implements IPeakFilter<ProcessPeaksByPeakClassifierFilterSettings> {
+
+	@Override
+	public String getName() {
+
+		return "Peak Classifier Filter";
+	}
+
+	@Override
+	public String getDescription() {
+
+		return "Filter peaks and add peak classifier (e.g. for whole name or parts of the name.)";
+	}
+
+	@Override
+	public Class<ProcessPeaksByPeakClassifierFilterSettings> getConfigClass() {
+
+		return ProcessPeaksByPeakClassifierFilterSettings.class;
+	}
+
+	@Override
+	public boolean acceptsIPeaks(Collection<? extends IPeak> items) {
+
+		return true;
+	}
+
+	@Override
+	public <X extends IPeak> void filterIPeaks(CRUDListener<X, IPeakModel> listener, ProcessPeaksByPeakClassifierFilterSettings configuration, MessageConsumer messageConsumer, IProgressMonitor monitor) throws IllegalArgumentException {
+
+		Collection<X> read = listener.read();
+		if(configuration == null) {
+			configuration = createConfiguration(read);
+		}
+		SubMonitor subMonitor = SubMonitor.convert(monitor, read.size());
+		LinkedHashMap<String, String> parsedUserDefinedValues = parseUserDefinedValuesAsMap(configuration);
+
+		for(X peak : read) {
+			setPeakClassifier(configuration, parsedUserDefinedValues, peak);
+			subMonitor.worked(1);
+		}
+	}
+
+	private static LinkedHashMap<String, String> parseUserDefinedValuesAsMap(ProcessPeaksByPeakClassifierFilterSettings configuration) {
+
+		String userDefinedMatchExpression = configuration.getUserDefinedMatchExpression();
+		String matchClassification = configuration.getMatchClassification();
+		List<List<String>> matchList = new ArrayList<List<String>>();
+		if(!userDefinedMatchExpression.isEmpty()) {
+			parseExpressions(matchList, userDefinedMatchExpression);
+		} else {
+			throw new IllegalArgumentException("No match expressions defined!");
+		}
+		if(!matchClassification.isEmpty()) {
+			parseExpressions(matchList, matchClassification);
+		} else {
+			throw new IllegalArgumentException("No match classifications defined!");
+		}
+		if(matchList.get(0).size() != matchList.get(1).size()) {
+			throw new IllegalArgumentException("The number of Expressions is not the same as the number of Classifications.");
+		}
+		LinkedHashMap<String, String> parsedUserDefinedValues = new LinkedHashMap<String, String>();
+		for(int i = 0; i < matchList.get(0).size(); i++) {
+			parsedUserDefinedValues.put(matchList.get(0).get(i),matchList.get(1).get(i));
+		}
+		return parsedUserDefinedValues;
+	}
+
+	private static void parseExpressions(List<List<String>> matchList, String userExpressions) {
```

Comment:
do not modify method paremeters inside the method. Better change return type from void to List<List<..